src/components/implementation/tests/unit_torrent/torrent_test.c

```diff
@@ -6,116 +6,223 @@
  */
 
 #include <cos_component.h>
+#include <stdio.h>
 #include <print.h>
 #include <sched.h>
 #include <cbuf.h>
 #include <evt.h>
 #include <torrent.h>
+#include <perf_test_read.h>
 
 //#define VERBOSE 1
 #ifdef VERBOSE
 #define printv(fmt,...) printc(fmt, ##__VA_ARGS__)
 #else
 #define printv(fmt,...) 
 #endif
+#define DEBUG 0
 
-char buffer[1024];
+long long doit(td_t t, int size)
+{
+	cbuf_t cb;
+	long long time1, time2;
+	int ret = 0;
+	char *s = cbuf_alloc(size, &cb); //null terminator?
 
-void cos_init(void)
+	memset(s, 'a', size - 2);
+	s[size - 1] = '\0';
+
+	if (!s) { printc("UNIT TEST FAILED: cbut alloc failed %s\n", s); return -1; }
+
+	/* take clock ticks */
+	rdtscll(time1);
+	ret = twritep(cos_spd_id(), t, cb, 0, size);
+	/* take clock ticks after */
+	rdtscll(time2);
+
+	long long diff = time2 - time1;	
+
+#if DEBUG
+	printc("t1: %lld t2: %lld diff: %lld\n", time1, time2, diff);
+	printc("ret - sz: %d\n", ret - size);
+#endif
+
+	return diff;
+}
+
+void test_write(void)
 {
-	td_t t1, t2;
-	long evt1, evt2;
+	long evt1 = 0;
+	td_t t;
 	char *params1 = "bar";
-	char *params2 = "foo/";
-	char *params3 = "foo/bar";
-	char *data1 = "1234567890", *data2 = "asdf;lkj", *data3 = "asdf;lkj1234567890";
-	unsigned int ret1, ret2;
+	int chunk_size;
+	
+	/* open file */
+	t = tsplit(cos_spd_id(), td_root, params1, strlen(params1), TOR_ALL, evt1);
+	if (t < 1) { printc("UNIT TEST FAILED: split failed %c\n", t); return; }
+
+	for (chunk_size = 64; chunk_size <= 16384; chunk_size *= 2) {
```

Comment:
style.  ` ; ` vs `; `  Small issue, but another indication that you want to pay some attention to style